src/client.coffee

```diff
@@ -331,9 +332,13 @@ class SlackClient
     if @eventHandler
       # fetch full representations of the user, bot, and potentially the item_user.
       fetches = {}
-      fetches.user = @fetchUser event.user if event.user
-      fetches.bot = @fetchBotUser event.bot_id if event.bot_id
-      fetches.item_user = @fetchUser event.item_user if event.item_user
+      if event.bot_id
+        fetches.bot = @fetchBotUser event.bot_id
+      else if event.user
```

Comment:
with this change, when the event has a `bot_id` we will not make a call to Web API method `users.info`, and therefore there won't be a value for `fetched.user` later. 

this seems safe because there's no place in the branch of code where `fetched.bot` is assigned, that the value of `fetched.user` is used. 👍

but i am wondering, how will this reduce the number of calls to `bots.info`?

Reply:
@aoberoi 
>but i am wondering, how will this reduce the number of calls to bots.info?

The root cause of the increase of `bots.info` calls is lack of the timing to run the following lines with the latest payloads.

* https://github.com/seratch/hubot-slack/blob/0a063513c4a8c7b53d6481d1b41d35f58622e208/src/client.coffee#L367
* https://github.com/seratch/hubot-slack/blob/0a063513c4a8c7b53d6481d1b41d35f58622e208/src/client.coffee#L372

The combination of the latest payloads and current `SlackClient` implementation keeps the `@botUserIdMap` empty in any case. So, `fetchBotUser` method calls `bots.info` API every time getting events with both `bot_id` and `user`.

Reply:
ohhh i see! So now that `event.user` is in every bot message, `fetched.user` always has a value, and then `botUserIdMap` is never updated. This change makes it so `fetched.user` will not have a value when both `event.bot` and `event.user` are present. 👍